samples/kotlin_compose/src/main/java/io/customer/android/sample/kotlin_compose/ui/login/LoginViewModel.kt

```diff
@@ -1,5 +1,113 @@
 package io.customer.android.sample.kotlin_compose.ui.login
 
+import android.util.Patterns
+import androidx.annotation.StringRes
 import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewModelScope
+import dagger.hilt.android.lifecycle.HiltViewModel
+import io.customer.android.sample.kotlin_compose.R
+import io.customer.android.sample.kotlin_compose.data.models.User
+import io.customer.android.sample.kotlin_compose.data.repositories.UserRepository
+import io.customer.sdk.CustomerIO
+import java.util.UUID
+import javax.inject.Inject
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.StateFlow
+import kotlinx.coroutines.flow.asStateFlow
+import kotlinx.coroutines.flow.update
+import kotlinx.coroutines.launch
+import kotlinx.coroutines.withContext
 
-class LoginViewModel : ViewModel()
+/**
+ * UI state for the Login screen
+ */
+data class LoginUiState(
+    @StringRes val emailError: Int? = null,
+    @StringRes val nameError: Int? = null,
+    val loading: Boolean = false,
+    val user: User? = null
+)
+
+@HiltViewModel
+class LoginViewModel @Inject constructor(
+    private val userRepository: UserRepository
+) : ViewModel() {
+
+    // UI state exposed to the UI
+    private val _uiState = MutableStateFlow(LoginUiState(loading = true))
+    val uiState: StateFlow<LoginUiState> = _uiState.asStateFlow()
+
+    init {
+        loadData()
+    }
+
+    fun loginUser(email: String, name: String, onLoginSuccess: () -> Unit) {
+        if (name.isEmpty()) {
+            _uiState.update {
+                it.copy(
+                    nameError = R.string.invalid_name,
+                    emailError = null
+                )
+            }
+        } else if (email.isEmpty() || Patterns.EMAIL_ADDRESS.matcher(email).matches().not()) {
+            _uiState.update {
+                it.copy(
+                    emailError = R.string.invalid_email,
+                    nameError = null
+                )
+            }
+        } else {
+            login(email = email, name = name, onLoginSuccess = onLoginSuccess)
+        }
+    }
+
+    private fun login(
+        email: String,
+        name: String,
+        isGuest: Boolean = false,
+        onLoginSuccess: () -> Unit
+    ) {
+        viewModelScope.launch(Dispatchers.IO) {
```

Comment:
Is this needed? Can we do everything on the main thread since our sample apps are not doing anything intensive?

Reply:
```
userRepository.login(email = email, name = name, isGuest = isGuest)
```

We are inserting things in DB and even though data isn't much i don't feel comfortable inserting stuff in DB in main thread